Makefile.coq.local

```diff
@@ -14,3 +14,26 @@ structures.vo : hb.elpi
 #  $(foreach M,$(shell seq 0 10),\
 #   $(foreach K,$(shell seq 0 10),\
 #     $(eval $(call MKRULE,$(N),$(M),$(K))))))
+
+pre-all:: hb
+
+hb: hb.ml
+	$(SHOW)'OCAMLC hb.ml -o hb'
+	$(HIDE) ocamlc unix.cma str.cma hb.ml -o hb
+
+ifneq "$(DESTDIR)" ""
```

Comment:
this is why it should work, something else is buggy

Reply:
Why didn't I have this in my local checkout, is it recent??

Reply:
ahah it works!

Reply:
But then I cannot use both the variables `COQMF_COQLIB` and `$DESTDIR` :-/

Reply:
I will hack around this...

Reply:
The behaviour of `DESTDIR` is superweird... why does it concatenate it with coq's full path...
```
$ make DESTDIR=$(mktemp -d) install
/nix/store/kvhw9snhf3hxnlfmqbkq88yf2bqk19iy-coq-8.13.0/bin/coq_makefile  -f _CoqProject -o Makefile.coq
make -f Makefile.coq  install
make[1] : on entre dans le répertoire « /home/cyril/git/hierarchy-builder/logging-vernac »
INSTALL structures.vo /run/user/1000/tmp.RVRxOFEV2d//nix/store/kvhw9snhf3hxnlfmqbkq88yf2bqk19iy-coq-8.13.0/lib/coq//user-contrib/HB/
INSTALL structures.v /run/user/1000/tmp.RVRxOFEV2d//nix/store/kvhw9snhf3hxnlfmqbkq88yf2bqk19iy-coq-8.13.0/lib/coq//user-contrib/HB/
INSTALL structures.glob /run/user/1000/tmp.RVRxOFEV2d//nix/store/kvhw9snhf3hxnlfmqbkq88yf2bqk19iy-coq-8.13.0/lib/coq//user-contrib/HB/
make[2] : on entre dans le répertoire « /home/cyril/git/hierarchy-builder/logging-vernac »
install hb /run/user/1000/tmp.RVRxOFEV2d/bin
make[2] : on quitte le répertoire « /home/cyril/git/hierarchy-builder/logging-vernac »
make[1] : on quitte le répertoire « /home/cyril/git/hierarchy-builder/logging-vernac »
```

Reply:
ah I think I figured it out!

Reply:
The right command is a combination of both `DESTDIR` and `COQMF_COQLIB`!!

Reply:
```
installFlags = [ "DESTDIR=$(out)" "COQMF_COQLIB=lib/coq/${coq.coq-version}" ];
```

Reply:
Can I rebase and push?

Reply:
please do.

It is not weird, it is the layout chose in nix which is not standard. `$PREFIX/lib/coq/$VERSION/stuff` if something made up by nix to have multiple coq versions coexist. Shall I repeat: *PUSH THESE CHANGE UPSTREAM* you nixer

Reply:
I will!

Reply:
`$PREFIX/lib/coq/$VERSION/stuff` I agree this is weird, it looks unnecessary to me... (none should ever install two versions of coq at the same time anyway...) I git blame and ask why.

Reply:
it may make sense, I don't know, but if it does it should not be done by hacking an internal coq_makefile variable

Reply:
you mean `COQMF_COQLIB` is supposed to be internal?

Reply:
Internal.... it is given by coq, it is not supposed to be overridden on the command line (it is not in the doc for example).

IMO, if you want to have coq populate lib/coq-$VERSION/ you can add a configure time option to it and have all coq projects get from coq the right path, a bit like `-prefix`.

Reply:
I think it may make sense to have this feature. I'm complaining because it is "simulated" via little hacks.

Reply:
Actually, you can already do it, there is a `-libdir`

Reply:
Nice!

Reply:
Mmmh, are you sure though...

Reply:
where did you find this?

Reply:
is it a configureflag?

Reply:
it is a flag in coq's configure

Reply:
I guess you can configure coq to use /usr/lib/coq-8.13/ and as a consequence coq_makefile will install things there without hacking COQMF_COQLIB (which contains the value at configure time for -coqlib) I guess.

Reply:
I don't thing it is a good idea, `COQLIB` should keep pointing to coq's main /lib/ directory, which we cannot overwrite...